src/syzygy/syzygy.h

```diff
@@ -212,8 +212,8 @@ inline bool generateSyzygyMoves(const BitBoard& board, movegen::ValidMoves& move
         if (!from.has_value() || !to.has_value())
             continue;
 
-        const auto piece = board.getPieceAtSquare(helper::positionToSquare(*from));
-        const auto target = board.getPieceAtSquare(helper::positionToSquare(*to));
+        const auto piece = board.getAttackerAtSquare(helper::positionToSquare(*from), board.player);
```

Comment:
Nitpick suggestion: Maybe we should try to stick to this term for "attackers" going forward?
```suggestion
        const auto attacker = board.getAttackerAtSquare(helper::positionToSquare(*from), board.player);
```